src/utils/ucc_math.h

```diff
@@ -66,4 +66,27 @@ int ucc_sort_uniq(int *array, int len, int inverse);
 
 #define ucc_div_round_up(_n, _d) (((_n) + (_d) - 1) / (_d))
 
+static inline float bfloat16tofloat32(const void *bfloat16_ptr)
+{
+    float res;
+    memset(&res, 0, sizeof(float));
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+    ((unsigned short *)(&res))[0] = *((unsigned short *)bfloat16_ptr);
+#else
+    ((unsigned short *)(&res))[1] = *((unsigned short *)bfloat16_ptr);
+#endif
+    return res;
+}
+
+static inline void float32tobfloat16(float float_val, void *bfloat16_ptr)
+{
+#if UCC_BIG_ENDIAN
+    *((unsigned short *)bfloat16_ptr) =
+            ((unsigned short *)(&float_val))[0];
```

Comment:
why not use uint16_t?

Reply:
@Sergei-Lebedev @vspetrov 
They did it in CUDA, in cuda_bf16.hpp... you think it's a mistake?

typedef struct __CUDA_ALIGN__(2) {
    unsigned short x;
} __nv_bfloat16_raw;

Reply:
from CUDA slack channel - "It is 2 bytes on all targets that CUDA currently supports"